### PR TITLE
refactor: move contact requests to new architecture

### LINKED ISSUE #4061: refactor: move contact requests to new architecture

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4061

Most of the contact request code was already moved, but it wasn't hooked to the QML yet and also there were missing events and some code to improve.

I think that with these changes, it would be possible to now remove the old code for contacts. I will do that in another PR

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]
nice!

**Comment by anastasiyaig** [comment]
@jrainville i created new account from scratch in the PR (commit https://github.com/status-im/status-desktop/commit/0fa49523f36530309184c11d776912580f5e41a4) and it looks like a lot of this are missing here:

can u please take a look:

flags enabling is not working:

Screenshot 2021-11-11 at 11 48 08


timeline and settings icons are the same
Screenshot 2021-11-11 at 11 47 55

when creating new account, there is a weird contact request to someone
Screenshot 2021-11-11 at 11 47 44

profile picture is not shown
Screenshot 2021-11-11 at 11 47 37

adding contact results in error

```
INF 2021-11-11 11:52:23.463+03:00 history request started                    topics="mailserver-interaction" tid=236337 file=core.nim:103
qrc:/StatusQ/src/StatusQ/Components/StatusRoundedImage.qml:27:5: QML Image: Cannot open: qrc:/app/AppLayouts/Profile/panels/(jsonrpc: %22%22, result: %22data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAgElEQVR4nOzWsQmAQBAFURV7sTI7sAw7sDKr0VwMVvC44TMvlEuGD4vTEMIQGkNoDKExhMYQGkNoDKGJCZmrD49zv96+r8s29nj/FLOIITSGqJGYRQyhMYTGfy0aQ2gMoSlfra+q1+YvMYsYQmOIGolZxBAaQ2gMoTGE5g4AAP//I/YZaHdYNaEAAAAASUVORK5CYII=%22, id: 0, error: nil)
```

Screenshot 2021-11-11 at 11 52 26

removed contact still appears in list when opening 1x1 chat menu

Screenshot 2021-11-11 at 11 53 53

**Comment by jrainville** [comment]
This is mostly a list for myself so I remember. Here are the issues that are part of `base_bc`, so were not introduced in my branch. I think they are fixed in master and once `base_bc` will be rebased, most of them will be fixed:
- timeline and setting as same icon
- New account contact request thingy
- profile picture not shown
- empty my profile

**Comment by anastasiyaig** [comment]
@jrainville ya, some of this is already fixed in master, maybe we need to rebase `base_bc` , but i think the issues about contact requests mentioned above should be relevant, at least i dont see them in master

**Comment by jrainville** [comment]
@anastasiyaig yeah all the contact issues should be from my own branch. I'll fix them now

**Comment by jrainville** [comment]
@anastasiyaig I fixed the contact issues. The other issues are caused by the base branch being very much a breaking change branch.

**Comment by anastasiyaig** [comment]
@jrainville were u able to finish fixing the issues for adding a contact? 

Screenshot 2021-11-15 at 13 20 30

**Comment by jrainville** [comment]
@anastasiyaig It should be fix. I just rebased and tested and it worked fine for me

**Comment by alexandraB99** [comment]
@jrainville is this ready for review? (it is marked as do-not-merge)

**Comment by jrainville** [comment]
@alexandraB99 yes. I think Nastya marked as Do not merge because she found a bug, but I fixed it, but also the code can still be reviewed.

File: ui/app/AppMain.qml [hunk 1]
```diff
@@ -55,19 +55,19 @@ Item {
             return profileModule.model.thumbnailImage
         }
 
-        const index = profileModel.contacts.list.getContactIndexByPubkey(pubkey)
+        const index = contactsModule.model.list.getContactIndexByPubkey(pubkey)
```
Comment (alexandraB99) [inline ui/app/AppMain.qml:58]: Could this be moved to AppMain's store? +all the below

**Review by alexandraB99 — CHANGES_REQUESTED** [review]

**Comment by jrainville** [comment]
@alexandraB99 thanks for the review. I focused more on the Nim side for the refactor. 
I know my store usage is subpar to say the least, but it was because I know Sale and some of you guys are already refactoring the QML side for chats and I wanted to be able to test quickly and possibly merge this without as many conflicts.

Let me know if you think I should still commit to refactoring the stores as well

**Comment by alexandraB99** [comment]
> @alexandraB99 thanks for the review. I focused more on the Nim side for the refactor. I know my store usage is subpar to say the least, but it was because I know Sale and some of you guys are already refactoring the QML side for chats and I wanted to be able to test quickly and possibly merge this without as many conflicts.
> 
> Let me know if you think I should still commit to refactoring the stores as well

fair enough, could you just please add TODO so that we move these parts in stores at some point of time?

**Comment by jrainville** [comment]
Good point. I'll add the TODOs once I rebase

**Comment by jrainville** [comment]
@alexandraB99 you can review again. I rebased and added TODOs.
I'll squash when I have your 👍

**Review by alexandraB99 — APPROVED** [review]

**Pull request merged**